src/main/java/com/couchbase/lite/Attachment.java

```diff
@@ -124,15 +125,36 @@ public String getContentType() {
     public InputStream getContent() throws CouchbaseLiteException {
         if (body != null) {
             return body;
-        }
-        else {
+        } else {
             Database db = revision.getDatabase();
-            Attachment attachment = db.getAttachmentForSequence(revision.getSequence(), this.name);
+            long sequence = getAttachmentSequence();
+            if (sequence == 0) {
+                throw new CouchbaseLiteException(Status.INTERNAL_SERVER_ERROR);
+            }
+            Attachment attachment = db.getAttachmentForSequence(sequence, this.name);
             body = attachment.getContent();
+            if (attachment.getGZipped()) {
```

Comment:
As you wrote, `body` instance variable should not be null in case the instance is created from `Database.getAttachmentForSequence`. Thanks.